llmfoundry/tokenizers/tiktoken.py

```diff
@@ -253,7 +253,10 @@ def _convert_token_to_id(self, token: str) -> Optional[int]:
 
     def _convert_id_to_token(self, index: int) -> Optional[str]:
         """Converts an index (integer) in a token (str) using the vocab."""
-        return self.decoder.get(index)
+        # For tokens in either the gap in ids in the tokenizer, or beyond the range of the tokenizer,
+        # we return empty string. This matches the behavior of Hugging Face fast tokenizers,
+        # but not slow tokenizers.
+        return self.decoder.get(index, '')
```

Comment:
nit: maybe add a warning if we hit an id gap or out of range index

Reply:
The other HF tokenizers don't do this so I think I'm going to leave it as is to match their behavior. Can add in a follow up if you feel strongly.

Reply:
nah that sounds good to leave as is